### ISSUE TITLE
出现循环引用

### ISSUE THREAD

**Issue by ricksuzade-maker** [immortalwrt/packages#22]
1806-k5.4，dee4f72版本，下载后准备编译，在defconfig的时候出现如下错误提示：
~/scripts/immortal/r2s/immortalwrt$ make defconfig
tmp/.config-package.in:110028:error: recursive dependency detected!
tmp/.config-package.in:110028:	symbol PACKAGE_ocserv depends on OCSERV_SECCOMP
feeds/packages/net/ocserv/Config.in:10:	symbol OCSERV_SECCOMP depends on PACKAGE_ocserv
For a resolution refer to Documentation/kbuild/kconfig-language.rst
subsection "Kconfig recursive dependency limitations"

#
# configuration written to .config
#

删除ocserv/Config.in里面对应段落后正常，不再出现以上错误，可是我担心这样会引起其他错误。

**Comment by zxlhhyccc** [comment]
> 1806-k5.4，dee4f72版本，下载后准备编译，在defconfig的时候出现如下错误提示：
> ~/scripts/immortal/r2s/immortalwrt$ make defconfig
> tmp/.config-package.in:110028:error: recursive dependency detected!
> tmp/.config-package.in:110028: symbol PACKAGE_ocserv depends on OCSERV_SECCOMP
> feeds/packages/net/ocserv/Config.in:10: symbol OCSERV_SECCOMP depends on PACKAGE_ocserv
> For a resolution refer to Documentation/kbuild/kconfig-language.rst
> subsection "Kconfig recursive dependency limitations"
> 
> # configuration written to .config
> 删除ocserv/Config.in里面对应段落后正常，不再出现以上错误，可是我担心这样会引起其他错误。
feeds/packages/libs/libseccomp/Makefile中将此行删除即可解决你的问题：
![image](https://user-images.githubusercontent.com/45259624/113469689-37566400-9482-11eb-8689-d23ae50eba85.png)

**Comment by 1715173329** [comment]
目前的一个 workaround：
```patch
diff --git a/net/ocserv/Makefile b/net/ocserv/Makefile
index ed05a3a..c9c4d65 100644
--- a/net/ocserv/Makefile
+++ b/net/ocserv/Makefile
@@ -39,7 +39,7 @@ define Package/ocserv
   TITLE:=OpenConnect VPN server
   URL:=http://www.infradead.org/ocserv/
   MAINTAINER:=Nikos Mavrogiannopoulos <n.mavrogiannopoulos@gmail.com>
-  DEPENDS:= +OCSERV_RADIUS:libradcli +OCSERV_HTTP_PARSER:libhttp-parser +OCSERV_SECCOMP:libseccomp +libgnutls +certtool +libncurses +libreadline +OCSERV_PAM:libpam +OCSERV_PROTOBUF:libprotobuf-c +libev +kmod-tun
+  DEPENDS:= +OCSERV_RADIUS:libradcli +OCSERV_HTTP_PARSER:libhttp-parser +(!arc&&OCSERV_SECCOMP):libseccomp +libgnutls +certtool +libncurses +libreadline +OCSERV_PAM:libpam +OCSERV_PROTOBUF:libprotobuf-c +libev +kmod-tun
   USERID:=ocserv=72:ocserv=72
 endef
```

**Comment by 1715173329** [comment]
Fixed in b0e621c78.
